omod/src/main/java/org/openmrs/module/idgen/rest/resource/IdentifierSourceResource.java

```diff
@@ -0,0 +1,303 @@
+        String name = validateInput(postBody.get("name"));
+        String description = validateInput(postBody.get("description"));
+
+        if (StringUtils.isBlank(identifierSourceType)) {
+            errors.add("source type"); 
+        }
+        if (StringUtils.isBlank(identifierTypeUuid) || patientIdentifierType == null) {
+            errors.add("patient identifier type"); 
+        }
+        if (StringUtils.isBlank(name)) {
+            errors.add("name"); 
+        }
+
+        if(identifierSourceType.equals("SequentialIdentifierGenerator")) {
+
+            SequentialIdentifierGenerator identifierSource = new SequentialIdentifierGenerator();
+            String firstIdentifierBase = validateInput(postBody.get("firstIdentifierBase"));
+            String baseCharacterSet = validateInput(postBody.get("baseCharacterSet"));
+            String prefix = validateInput(postBody.get("prefix"));
+            String suffix = validateInput(postBody.get("suffix"));
+            String minLength = validateInput(postBody.get("minLength"));
+            String maxLength = validateInput(postBody.get("maxLength"));
+            
+            if (StringUtils.isBlank(firstIdentifierBase)) { 
+                errors.add("first identifier base"); 
+            }
+            if (StringUtils.isBlank(baseCharacterSet)) { 
+                errors.add("base character set"); 
+            }
+            if (!StringUtils.isBlank(description)) {
+                identifierSource.setDescription(description);
+            }
+            if (!StringUtils.isBlank(prefix)) { 
+                identifierSource.setPrefix(prefix); 
+            }
+            if (!StringUtils.isBlank(suffix)) { 
+                identifierSource.setSuffix(suffix); 
+            }
+            if (!StringUtils.isBlank(minLength)) {
+                if(StringUtils.isNumeric(minLength)) {
+                        identifierSource.setMinLength(Integer.parseInt(minLength));
+                }else {
+                        errors.add("min length");
+                }
+            }
+            if (!StringUtils.isBlank(maxLength)) {
+                if(StringUtils.isNumeric(maxLength)) {
+                        identifierSource.setMaxLength(Integer.parseInt(maxLength));
+                }else {
+                        errors.add("max length");
+                }
+            }
+            if(errors.toString().length() > 2) {
+                throw new ConversionException("The values of the following inputs are missing or invalid: " + errors.toString());
+            }
+
+            identifierSource.setIdentifierType(patientIdentifierType);
+            identifierSource.setName(name);
+            identifierSource.setDescription(description);
+            identifierSource.setBaseCharacterSet(baseCharacterSet);
+            identifierSource.setFirstIdentifierBase(firstIdentifierBase);
+
+            savedIdentifierSource =  save(identifierSource);	
+            }
```

Comment:
The above closing brace is not indented properly

Reply:
fixed